dropwizard-logging/src/main/java/io/dropwizard/logging/DefaultLoggingFactory.java

```diff
@@ -167,12 +172,59 @@ public void stop() {
         // Should acquire the lock to avoid concurrent listener changes
         CHANGE_LOGGER_CONTEXT_LOCK.lock();
         try {
+            // We need to go through a list of appenders and locate the async ones,
+            // as those could have messages left to write. Since there is no flushing
+            // mechanism built into logback, we wait for a short period of time before
+            // giving up that the appender will be completely flushed.
+            final Logger logger = loggerContext.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+            final ArrayList<Appender<ILoggingEvent>> appenders = Lists.newArrayList(logger.iteratorForAppenders());
+            for (Appender<ILoggingEvent> appender : appenders) {
+                if (appender instanceof AsyncAppender) {
+                    flushAppender((AsyncAppender) appender);
+                }
+            }
+        } catch (InterruptedException ignored) {
+            // If the thread waiting for the logs to be flushed is aborted then
```

Comment:
I think it always a good practice, because we generally don't control the thread which loads `DefaultServerFactory`. There is a lot of people who use Dropwizard within app servers.